tests/Aspire.Hosting.Tests/Redis/AddRedisTests.cs

```diff
@@ -256,10 +256,16 @@ public void WithDataVolumeAddsPersistenceAnnotation()
         var redis = builder.AddRedis("myRedis")
                               .WithDataVolume();
 
-        var persistenceAnnotation = redis.Resource.Annotations.OfType<RedisPersistenceCommandLineArgsCallbackAnnotation>().Single();
+        Assert.True(redis.Resource.TryGetAnnotationsOfType<CommandLineArgsCallbackAnnotation>(out var argsCallbacks));
 
-        Assert.Equal(TimeSpan.FromSeconds(60), persistenceAnnotation.Interval);
-        Assert.Equal(1, persistenceAnnotation.KeysChangedThreshold);
+        var args = new List<object>();
+        foreach (var argsAnnotation in argsCallbacks)
+        {
+            Assert.NotNull(argsAnnotation.Callback);
+            argsAnnotation.Callback(new CommandLineArgsCallbackContext(args));
+        }
+
+        Assert.Equal("--save 60 1".Split(" "), args);
```

Comment:
I like testing like this. I think we should keep this change, it actually tests more like a customer. Call the API, and observe the change in behavior.